src/pages/index.js

```diff
@@ -54,43 +64,213 @@ const addFormValidator = new FormValidator(
+  profileEditSaveButton.disabled = true;
+  api
+    .updateUserInfo(data["edit-profile"], data["profile-description"])
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      profileEditPopup.close();
+      profileEditSaveButton.textContent = "Save";
+      profileEditSaveButton.classList.remove("loading");
+      profileEditSaveButton.disabled = false;
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    });
 }
 
+const handleLikeClick = (card) => {
+  api
+    .toggleLike(card.getId(), card.isLiked())
+    .then(() => {
+      card.updateLikesView();
+    })
+    .catch((err) =>
+      console.error(`An error occurred when changing like status: ${err}`)
+    );
+};
+
 function handleAddCardFormSubmit(data) {
-  const newCardElement = createCard({ name: data.title, link: data.url });
-  section.addItem(newCardElement);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-  addCardPopup.close();
+  addCardSaveButton.textContent = "Saving...";
+  addCardSaveButton.classList.add("loading");
+  addCardSaveButton.disabled = true;
+  const sanitizedUrl = data.url.replace(/["'\s]+$/, "").trim();
+  api
+    .addCard(data.title, sanitizedUrl)
+    .then((newCardInfo) => {
+      const newCardElement = createCard({
+        name: newCardInfo.name,
+        link: newCardInfo.link,
+        _id: newCardInfo._id,
+      });
+      section.addItem(newCardElement);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+      addCardSaveButton.textContent = "Save";
+      addCardSaveButton.classList.remove("loading");
+      addCardSaveButton.disabled = false;
+    })
+    .catch((err) => {
+      console.error("Error adding new card:", err);
+    });
+}
+
+function handleDeleteConfirm(data) {
+  const cardId = data.cardId;
+  if (!cardId || typeof cardId !== "string") {
+    console.error(
+      "Invalid cardId in handleDeleteConfirm:",
+      cardId,
+      "Data:",
+      data
+    );
+    return;
+  }
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      section.removeItem(cardId);
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+function handleAvatarFormSubmit(data) {
+  avatarSubmitButton.textContent = "Saving...";
+  avatarSubmitButton.classList.add("loading");
+  avatarSubmitButton.disabled = true;
+  api
+    .updateUserAvatar(data.avatar)
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      avatarEditPopup.close();
+      avatarSubmitButton.textContent = "Save";
+      avatarSubmitButton.classList.remove("loading");
+      avatarSubmitButton.disabled = false;
+    })
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.